packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect, useCallback, useRef, useLayoutEffect } from 'react';
+import { Text, Padded } from '@buffetjs/core';
+import { LoadingIndicator } from 'strapi-helper-plugin';
+import PropTypes from 'prop-types';
+
+import getRequestUrl from '../../utils/getRequestUrl';
+import getMockData from './mockData';
+import { useListView } from '../../hooks';
+import Tooltip from '../Tooltip';
+
+const RelationPreviewTooltip = ({ tooltipId, rowId, mainField, name }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [relationData, setRelationData] = useState([]);
+  const { slug } = useListView();
+  const tooltipRef = useRef();
+
+  const fetchRelationData = useCallback(async () => {
+    const requestURL = getRequestUrl(`collection-types/${slug}/${rowId}/${name}`);
+    try {
+      // TODO : Wait for the API
+      // const { data } = await request(requestURL, {
+      //   method: 'GET',
+      // });
+
+      console.log(requestURL);
+      setRelationData(getMockData(mainField));
+      setIsLoading(false);
+    } catch (err) {
+      console.error({ err });
+      setIsLoading(false);
+    }
+  }, [mainField, name, rowId, slug]);
+
+  useEffect(() => {
+    // temp : Should remove the setTimeout and fetch the data
+    const timeout = setTimeout(() => {
```

Comment:
Can you just add the cancel subscriptions code even though we are not fetching data yet?

```
const abortController = new AbortController();
// ...

return () => {
  clearTimeout(timeout);
  abortController.abort();
}
```